src/Plugin/Block/RepeatSchedulesBlock.php

```diff
@@ -121,17 +121,15 @@ public function getLocations() {
    * @return array
    */
   public function getCategories(array $nids) {
-    $config = \Drupal::service('config.factory')->get('openy_gxp.settings');
     $query = $this->database->select('node_field_data', 'nfd');
-    $query->leftJoin('node__field_activity_category', 'sc', 'sc.entity_id = nfd.nid');
-    $query->condition('sc.field_activity_category_target_id', $config->get('activity'));
     $query->fields('nfd', ['title']);
     $query->condition('type', 'activity');
     $query->condition('status', 1);
     if ($nids) {
       $query->condition('nid', $nids, 'NOT IN');
     }
     $query->orderBy('title');
+    $query->distinct(TRUE);
```

Comment:
would be great to have Drupal->logger->warning("We have duplicates here, please do cleanup action")

Reply:
@podarok we're not currently testing if there are duplicates or not, and the `distinct` is set without a condition. I think what you're suggesting would require some additional code like:

```
$query->execute()
if 
  $results have duplicates
then 
  log->warning
  $query->distinct
  $query->execute
```

That seems possibly overkill to me?